types/src/key.rs

```diff
@@ -236,6 +251,12 @@ impl Key {
             Key::URef(uref) => uref.addr(),
             Key::Transfer(transfer_addr) => transfer_addr.value(),
             Key::DeployInfo(addr) => addr.value(),
+            Key::AuctionInfo(era_id) => {
+                // stop-gap measure until this method is removed
+                let mut ret = [0u8; BLAKE2B_DIGEST_LENGTH];
+                ret.clone_from_slice(&era_id.to_le_bytes());
+                ret
```

Comment:
This will panic as `BLAKE2B_DIGEST_LENGTH != era_id.to_le_bytes().len()`

My suggestion would be to use big endian and pad to right (but that's subjective - copying into first 8 bytes would work too):

```suggestion
                let mut ret = [0u8; BLAKE2B_DIGEST_LENGTH];
                let era_id_bytes = era_id.to_be_bytes();
                ret[BLAKE2B_DIGEST_LENGTH - era_id_bytes.len()..].copy_from_slice(&era_id_bytes);
                ret
```

Reply:
Darn, thought it might work.

Reply:
New approach:
```diff
diff --git a/types/src/key.rs b/types/src/key.rs
index 489975a0..0571bf22 100644
--- a/types/src/key.rs
+++ b/types/src/key.rs
@@ -254,7 +254,12 @@ impl Key {
             Key::AuctionInfo(era_id) => {
                 // stop-gap measure until this method is removed
                 let mut ret = [0u8; BLAKE2B_DIGEST_LENGTH];
-                ret.clone_from_slice(&era_id.to_le_bytes());
+                let era_id_bytes = era_id.to_le_bytes();
+                let era_id_bytes_len = era_id_bytes.len();
+                assert!(era_id_bytes_len < BLAKE2B_DIGEST_LENGTH);
+                for i in 0..era_id_bytes_len {
+                    ret[i] = era_id_bytes[i]
+                }
                 ret
             }
         }
```